src/State.ml

```diff
@@ -11,35 +11,25 @@ end
 
 module Make (State : Sigs.Type) =
 struct
-  type _ Effect.t +=
-    | Get : State.t Effect.t
-    | Set : State.t -> unit Effect.t
+  type _ eff +=
+    | Get : State.t eff
+    | Set : State.t -> unit eff
 
   let get () = Effect.perform Get
   let set st = Effect.perform (Set st)
 
   let run ~(init:State.t) f =
     let open Effect.Deep in
     let st = ref init in
-    try_with f ()
-      { effc = fun (type a) (eff : a Effect.t) ->
-            match eff with
-            | Get -> Option.some @@ fun (k : (a, _) continuation) ->
-              continue k !st
-            | Set v -> Option.some @@ fun (k : (a, _) continuation) ->
-              st := v; continue k ()
-            | _ -> None }
+    try f () with
+    | effect Get, k -> continue k !st
+    | effect Set v, k -> st := v; continue k ()
 
   let try_with ?(get=get) ?(set=set) f =
     let open Effect.Deep in
```

Comment:
Is `open Effect.Deep` needed anymore?

Reply:
Yes. See https://github.com/ocaml/ocaml/issues/13511 for my Thought.

Reply:
Wow, that is a very nice Thought.

Reply:
@jonsterling It's also why we have these two helper functions:

https://github.com/RedPRL/algaeff/blob/main/src/Fun.ml

I will be great if the compiler can do it directly, avoiding the overhead of creating stupid closures.